app/models/user.rb

```diff
@@ -0,0 +1,11 @@
+class User < ApplicationRecord
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i.freeze
+
+  before_save{email.downcase!}
+  validates :name, presence: true, length: {maximum: 50}
+  validates :email, presence: true, length: {maximum: 255},
+                    format: {with: VALID_EMAIL_REGEX},
+                    uniqueness: true
+  has_secure_password
```

Comment:
giải thích line code này (9)

Reply:
+ Lưu field password vào cơ sở dữ liệu dưới dạng đã được mã hóa .
+ Tạo ra 2 thuộc tính ảo password và pasword_confirmation .2 thuộc tính này not null và yêu cầu chúng phải trùng khớp
+ Tạo ra phương thức authenticate trả về thông tin người dùng khi mật khẩu chính xác